ts/main.ts

```diff
@@ -0,0 +1,140 @@
+const $locationInput = document.querySelector(
+  'input[type="text"]',
+) as HTMLInputElement;
+const $cityName = document.querySelector('.city-name') as HTMLHeadingElement;
+const $weatherDescription = document.querySelector(
+  '.weather-description',
+) as HTMLParagraphElement;
+const $weatherImg = document.querySelector('.weather-img') as HTMLImageElement;
+const $temperature = document.querySelector(
+  '.temperature',
+) as HTMLHeadingElement;
+const $humidity = document.querySelector('.humidity') as HTMLParagraphElement;
+const $windSpeed = document.querySelector(
+  '.wind-speed',
+) as HTMLParagraphElement;
+const $weatherDataDiv = document.getElementById(
+  'weather-data',
+) as HTMLDivElement;
+const $weatherForm2 = document.querySelector(
+  '#weather-form2',
+) as HTMLFormElement;
+const $locationInput2 = document.querySelector(
+  '#location-input2',
+) as HTMLInputElement;
+const $backButton = document.querySelector('#back-button') as HTMLAnchorElement;
+const $background = document.querySelector('body') as HTMLBodyElement;
+
+if (!$background) throw new Error('background query failed');
+if (!$backButton) throw new Error('back button query failed');
+if (!$locationInput2) throw new Error('Weather input 2 query failed');
+if (!$weatherForm2) throw new Error('Weather form 2 query failed');
+if (!$weatherDisplay) throw new Error('weather display query failed');
+if (!$weatherDataDiv) throw new Error('weather data div query failed');
+if (!$form) throw new Error('form query failed');
+if (!$locationInput) throw new Error('location input query failed');
+if (!$cityName) throw new Error('city name query failed');
+if (!$weatherDescription) throw new Error('weather description query failed');
+if (!$weatherImg) throw new Error('weather image query failed');
+if (!$temperature) throw new Error('temperature query failed');
+if (!$humidity) throw new Error('humidity query failed');
+if (!$windSpeed) throw new Error('wind speed query failed');
+
+const apiKey = 'b03d66ceebc0a945c3eb2bb9cc3551d1';
+
+$form.addEventListener('submit', async (event: Event) => {
+  event.preventDefault();
+  const location = $locationInput.value;
+  await fetchData(location);
+});
+
+$weatherForm2.addEventListener('submit', async (event: Event) => {
+  event.preventDefault();
+  const location = $locationInput2.value;
+  await fetchData(location);
+});
+
+$backButton.addEventListener('click', () => {
+  swapView('weather-app');
+});
+
+async function fetchData(location: string): Promise<void> {
+  try {
+    const response = await fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${location}&appid=${apiKey}&units=imperial`,
+    );
+    if (!response.ok) {
+      throw new Error('HTTP Error');
+    }
+    const data = await response.json();
+    displayWeatherData(data);
+    swapView('weather-display');
+    $locationInput.value = '';
+  } catch (error) {
+    console.log('Error:', error);
+  }
+}
```

Comment:
You should change this `console.log('Error:', error)` to `console.error('Error:', error)`